consensus/src/aggregation/safe_tip.rs

```diff
@@ -0,0 +1,247 @@
+use super::types::Index;
+use commonware_utils::{max_faults, Array};
+use std::collections::{btree_map, BTreeMap, HashMap, HashSet};
+
+/// A data structure that keeps track of the reported tip for each validator.
```

Comment:
This is an awesome data structure.

Reply:
Wonder if it makes sense to give it a catchier name given so (and publicly export it maybe in `utils`)?